java-checks/src/main/java/org/sonar/java/checks/security/XmlExternalEntityProcessingCheck.java

```diff
@@ -183,7 +196,35 @@ private static MethodMatcher setFeatureMethodMatcher(String className) {
       return MethodMatcher.create()
         .typeDefinition(subtypeOf(className))
         .name("setFeature")
-        .parameters("java.lang.String", "boolean");
+        .parameters(JAVA_LANG_STRING, "boolean");
+    }
+  }
+
+  private static class AccessExternalDTDOrSchemaPredicate implements Predicate<MethodInvocationTree> {
+
+    private final MethodMatcher methodMatcher;
+
+    private AccessExternalDTDOrSchemaPredicate(String className) {
+      this.methodMatcher = setPropertyMethodMatcher(className);
+    }
+
+    @Override
+    public boolean test(MethodInvocationTree methodInvocation) {
+      if (methodMatcher.matches(methodInvocation)) {
+        Arguments arguments = methodInvocation.arguments();
+        String propertyName = resolveAsStringConstant(arguments.get(0));
+        String propertyValue = resolveAsStringConstant(arguments.get(1));
```

Comment:
If you use this resolveStringAsConstant I tend to think there should be testcase with the inlined value of the property (ie not using the constant).

Reply:
good point, I will add a test case for that.